### PR TITLE
Allow passing options to exec call

### PR DESCRIPTION
**Pull request by htaunay**
In a particular scenario when working with node-rpm-builder I needed to increase the max buffer for the exec call, something like this:

`exec(cmd, {maxBuffer: 1024 *1024}, function rpmbuild(err, stdout) { ... } );`

I figured that maybe other users could eventually be interested in passing different options to the exec call, so I kept it generic.

In case you are interested, the exact problem was this: http://stackoverflow.com/questions/23429499/stdout-buffer-issue-using-node-child-process

### CODE REVIEW

**Comment by rictorres** [comment]
Good catch!

But in order to allow the user to pass these `exec` options, we need to receive it in the exposed method (`module.exports`).

Perhaps something like this?

```js
var buildRpm = require('rpm-builder');

var options = {
  name: 'my-project',
  version: '0.0.0',
  release: '1',
  buildArch: 'noarch',
  files: [],
  execOpts: {
    maxBuffer: 1024 * 1024
  }
};

buildRpm(options, function(err, rpm) {
  // ...
});
```

Then on `index.js`:
Line [163](https://github.com/rictorres/node-rpm-builder/blob/f54c4d1af9f443c33906fe3d73f0a05f2a05abc8/index.js#L163).
```js
keepTemp: false,
execOpts: {}
```

Line [188](https://github.com/rictorres/node-rpm-builder/blob/f54c4d1af9f443c33906fe3d73f0a05f2a05abc8/index.js#L188).
```js
buildRpm(buildRoot, specFile, options.rpmDest, options.execOpts, function(err, rpm) {
  // ...
});
```
Since we are passing `options.execOpts` before the `callback`, we need to update [this line](https://github.com/rictorres/node-rpm-builder/pull/12/files#diff-168726dbe96b3ce427e7fedce31bb0bcR103):
```js
function buildRpm(buildRoot, specFile, rpmDest, options, cb) {
  // ...
});
```

What do you think? I'm open to suggestions :)

**Comment by htaunay** [comment]
Looks good to me!
I just noticed that my change didn't actually change anything ... I had made a hard-coded change in my version and expected it to work. Sorry bout that

Do you want me to make the changes, or do you prefer to do it yourself?

**Comment by rictorres** [comment]
No worries!
Can you make them? Would be better if you did.
Then everything will be done in your PR.

Thanks, man!

**Comment by htaunay** [comment]
Done. Basically, if I am not mistaken, you had already specified all the necessary steps.
I did the following test:

```javascript
var options = { /* hash without exec options */ };
buildRpm(options, function(error, rpm) {...} );
options.execOpts = { maxBuffer: 1024 * 1024};
buildRpm(options, function(error, rpm) {...} );
```

In the first case the build would crash, and in the second case everything went fine.
BTW, the Travis test hasn't passed because it claims that when running with node v0.12.0 the test never ends... However, [when looking at the log directly](https://travis-ci.org/rictorres/node-rpm-builder/jobs/55652650) its seems clear that the test has in fact passed. Or am I missing something?

Thanks in advance!

**Comment by rictorres** [comment]
Nice!

Just one more thing, on your last commit I think you added two extra files by mistake: `.index.js.swp` and `test/.main.js.swp`. Can you check that?

File: index.js [hunk 1]
```diff
@@ -113,7 +113,10 @@ function buildRpm(buildRoot, specFile, rpmDest, cb) {
 
   logger(chalk.cyan('Executing:'), cmd);
 
-  exec(cmd, {}, function rpmbuild(err, stdout) {
+  if(execOpts === undefined)
+    execOpts = {};
```
Comment (rafaelverger) [inline index.js]: could be `execOpts = execOpts || {};`

**Comment by htaunay** [comment]
Whoops! [facepalm]
Swap files removed and Rafael's suggestion included

PS: As of the time of this comment the Travis build has yet begun, still waiting for the checks to be 100%

**Comment by rictorres** [comment]
All green! I'm gonna merge soon :)

Thanks again, @htaunay

**Pull request merged**